src/main/java/org/anarres/parallelgzip/ParallelGZIPOutputStream.java

```diff
@@ -71,41 +78,47 @@ public Block(@Nonnegative int index) {
          */
         // Only on worker thread
         @Override
-        public byte[] call() throws Exception {
+        public Block call() throws IOException {
             // LOG.info("Processing " + this + " on " + Thread.currentThread());
 
             State state = STATE.get();
             // ByteArrayOutputStream buf = new ByteArrayOutputStream(in.length);   // Overestimate output size required.
             // DeflaterOutputStream def = newDeflaterOutputStream(buf);
             state.def.reset();
             state.buf.reset();
-            state.str.write(in, 0, in_length);
+            state.str.write(buf, 0, buf_length);
             state.str.flush();
 
+            buf_length = state.buf.size();
+            assert buf_length <= buf.length;
+            state.buf.writeTo(buf);
```

Comment:
This re-use of the array is the crux, I think, because it lets us store the result without allocating a new byte[].